mantidproject/mantid
# IntegrateEllipsoids share background with satellite peaks

**Description of work.**

This adds the `ShareBackground` option to `IntegrateEllipsoids` to allow satellite peaks to use the same background as Bragg peaks. Satellite peaks can still have their peak intensities integrated separately, but they can now re-use the background of the main peak when subtracting background.

**To test:**



Verify the IntegrateEllipsoids tests and refer to the [testing script posted in the story](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/370). Run the script with and without the `ShareBackground` option and verify that the background shell for satellite peaks is the same as its corresponding bragg peak. Check that the peak shell, intensity, and sigma intensity is being set properly based on the shared background.

*There is no associated issue.*





Version into `ornl-next`: #32228 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
